axonserver/src/main/java/io/axoniq/axonserver/localstorage/file/WritePosition.java

```diff
@@ -12,35 +12,53 @@
 import java.util.Comparator;
 
 /**
+ * Describes a block where the event store can write a transaction.
+ *
  * @author Marc Gathier
  */
 public class WritePosition implements Comparable<WritePosition> {
 
-    static final WritePosition INVALID = new WritePosition(Long.MAX_VALUE, Integer.MAX_VALUE, null, null);
+    static final WritePosition INVALID = new WritePosition(Long.MAX_VALUE, Integer.MAX_VALUE, null, null, 0);
     private static final Comparator<WritePosition> writePositionComparator =
             Comparator.comparingLong(WritePosition::getSegment)
                       .thenComparingInt(WritePosition::getPosition);
     final long sequence;
     final int position;
     final WritableEventSource buffer;
     final Long segment;
+    public int prevEntries;
 
-    public WritePosition(long sequence, int position) {
-        this(sequence, position, null, null);
+    public WritePosition(long prevEntries, int position) {
```

Comment:
This constructor allows the creation of an inconsistent WritePosition and should be deleted.